lib/Net/Kubernetes/Role/ResourceCatalog.pm

```diff
@@ -0,0 +1,76 @@
+package Net::Kubernetes::Role::ResourceCatalog;
+
+use Moose::Role;
+
+requires 'server_version';
+requires 'url';
+
+my %endpoint_catalog = (
+	1.5 => {
+		pod                   => 'api/v1',
+		replicationcontroller => 'api/v1',
+		service               => 'api/v1',
+		event                 => 'api/v1',
+		node                  => 'api/v1',
+		serviceaccount        => 'api/v1',
+		secret                => 'api/v1',
+		endpoint              => 'api/v1',
+		deployment            => 'apis/extensions/v1beta1',
+		replicaset            => 'apis/extensions/v1beta1',
+	},
+	1.6 => {
+		pod                   => 'api/v1',
+		replicationcontroller => 'api/v1',
+		service               => 'api/v1',
+		event                 => 'api/v1',
+		node                  => 'api/v1',
+		serviceaccount        => 'api/v1',
+		secret                => 'api/v1',
+		endpoint              => 'api/v1',
+		deployment            => 'apis/apps/v1beta1',
+		replicaset            => 'apis/extensions/v1beta1',
+	},
+);
+
+my %globals_catalog = (
+	node           => 1,
+	serviceaccount => 1,
+);
+
+=method resource_path($resource_name) 
+
+Returns the full path of resources for the given version of kubernetes.
+
+=cut
+
+sub get_resource_path {
+	my ($self, $resource) = @_;
+	my $url            = $self->url;
+	my $server_version = $self->server_version;
+
+	$resource          = lc($resource);
+	my $resource_api   = $endpoint_catalog{$server_version}{$resource} || 'api/v1';
```

Comment:
I feel like we should find the previous version if a version is higher than the highest we know about, because that is likely to be the closest match.  So server 1.7 would get the 1.6 urls until you can update it with any changes for 1.7. Either that or error with unsupported version or something if the server version is higher than you expect.